python_modules/event-pipeline-demo/event_pipeline_demo/utils.py

```diff
@@ -0,0 +1,48 @@
+import re
+import subprocess
+
+from threading import Thread
+from six.moves import queue
+
+
+def run_spark_subprocess(cmd, logger):
+    """See https://bit.ly/2OpksJC for source of the subprocess stdout/stderr capture pattern in this
+    function.
+    """
+
+    # Spark sometimes logs in log4j format. In those cases, we detect and parse
+    log4j_regex = r'^(\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}) ([A-Z]{3,5})(.*?)$'
```

Comment:
can we point to the source of this regex?

Reply:
yep, added a comment. The best part about Spark is that it doesn't consistently strip newlines from things, so some log lines don't have the log4j prefix, e.g. parquet schema: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetWriteSupport.scala#L103-L108